front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -35,39 +38,43 @@ export default function MapSearchStation(props) {
 
   // Create playload based on the type of search "name" or "trigram"
 
-  const createPayload = (searchQuery) => ({
+  const createPayload = (searchQuery: (string | string[])[]) => ({
     object: 'operationalpoint',
-    query: ['and', searchQuery, ['=', ['infra_id'], infraID]],
+    query: ['and', searchQuery, ['=', ['infra_id'], infraID]] as SearchQuery,
   });
 
   // Sort on name, and on yardname
-  const orderResults = (results) =>
+  const orderResults = (results: IStationSearchResult[]) =>
     results.sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
 
   const searchByTrigrams = useCallback(async () => {
     const searchQuery = ['=i', ['trigram'], searchState];
     const payload = createPayload(searchQuery);
-    const { data, error } = await postSearch({
+    await postSearch({
       body: payload,
-    });
-    if (error) {
-      resetSearchResult();
-    } else {
-      setTrigramResults(data);
-    }
+    })
+      .unwrap()
+      .then((results) => {
+        setTrigramResults(results as IStationSearchResult[]);
+      })
+      .catch(() => {
+        resetSearchResult();
+      });
   }, [searchState]);
 
   const searchByNames = useCallback(async () => {
     const searchQuery = ['search', ['name'], searchState];
     const payload = createPayload(searchQuery);
-    const { data, error } = await postSearch({
+    await postSearch({
```

Comment:
It would be better to fix the openAPi file to have the correct `PostSearchApiResponse` type generated in `osrdEditoastApi.ts`, rather than creating the type manually on our side. (the `IStationSearchResult` you created)

Reply:
agree

Reply:
Done !